src/main/java/com/snowflake/kafka/connector/internal/BufferThreshold.java

```diff
@@ -123,19 +125,26 @@ public long getFlushTimeThresholdSeconds() {
    * @param ingestionMethodConfig ingestion method used. Check {@link IngestionMethodConfig}
    * @return true if all thresholds are valid.
    */
-  public static boolean validateBufferThreshold(
+  public static Map<String, String> validateBufferThreshold(
       Map<String, String> providedSFConnectorConfig, IngestionMethodConfig ingestionMethodConfig) {
-    return verifyBufferFlushTimeThreshold(providedSFConnectorConfig, ingestionMethodConfig)
-        && verifyBufferCountThreshold(providedSFConnectorConfig)
-        && verifyBufferBytesThreshold(providedSFConnectorConfig);
+    Map<String, String> invalidConfigParams = new HashMap<>();
+    invalidConfigParams.putAll(
+        verifyBufferFlushTimeThreshold(providedSFConnectorConfig, ingestionMethodConfig));
+    invalidConfigParams.putAll(verifyBufferCountThreshold(providedSFConnectorConfig));
+    invalidConfigParams.putAll(verifyBufferBytesThreshold(providedSFConnectorConfig));
+    return invalidConfigParams;
   }
 
-  private static boolean verifyBufferFlushTimeThreshold(
+  private static Map<String, String> verifyBufferFlushTimeThreshold(
       Map<String, String> providedSFConnectorConfig, IngestionMethodConfig ingestionMethodConfig) {
+    Map<String, String> invalidConfigParams = new HashMap();
+
     if (!providedSFConnectorConfig.containsKey(
         SnowflakeSinkConnectorConfig.BUFFER_FLUSH_TIME_SEC)) {
-      LOGGER.error("Config {} is empty", SnowflakeSinkConnectorConfig.BUFFER_FLUSH_TIME_SEC);
-      return false;
+      invalidConfigParams.put(
+          SnowflakeSinkConnectorConfig.BUFFER_FLUSH_TIME_SEC,
+          Utils.formatString(
+              "Config {} is empty", SnowflakeSinkConnectorConfig.BUFFER_FLUSH_TIME_SEC));
```

Comment:
Given previous behavior, could technically return a string instead of a map here. However I think it is better to do a map in case we have further param testing in the future. Open to changing it though